src/csharp/Microsoft.Spark/Sql/Row.cs

```diff
@@ -126,26 +106,9 @@ public override string ToString()
         /// </summary>
         /// <param name="obj">Other object to compare against</param>
         /// <returns>True if the other object is equal.</returns>
-        public override bool Equals(object obj)
-        {
-            if (obj is null)
-            {
-                return false;
-            }
-
-            if (ReferenceEquals(this, obj))
-            {
-                return true;
-            }
-
-            if (obj is Row otherRow)
-            {
-                return Values.SequenceEqual(otherRow.Values) &&
-                    Schema.Equals(otherRow.Schema);
-            }
-
-            return false;
-        }
+        public override bool Equals(object obj) =>
+            ReferenceEquals(this, obj) ||
+            ((obj is Row row) && Values.SequenceEqual(row.Values)) && Schema.Equals(row.Schema);
```

Comment:
`_genericRow.Equals(row._genericRow)` instead of `Values.SequenceEqual(row.Values))`